vcap_utils/vcap_utils/backends/base_tensorrt.py

```diff
@@ -0,0 +1,278 @@
+import numpy as np
+
+import pycuda.driver as cuda
+import tensorrt as trt
+import cv2
+
+from typing import Dict, List, Tuple, Optional, Any
+
+from vcap import (
+    Resize,
+    DETECTION_NODE_TYPE,
+    OPTION_TYPE,
+    BaseStreamState,
+    BaseBackend,
+    rect_to_coords,
+    DetectionNode,
+)
+
+
+class HostDeviceMem(object):
+    def __init__(self, host_mem, device_mem):
+        self.host = host_mem
+        self.device = device_mem
+
+    def __str__(self):
+        return "Host:\n" + str(self.host) + "\nDevice:\n" + str(self.device)
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class AllocatedBuffer:
+    def __init__(self, inputs_, outputs_, bindings_, stream_):
+        self.inputs = inputs_
+        self.outputs = outputs_
+        self.bindings = bindings_
+        self.stream = stream_
+
+
+class BaseTensorRTBackend(BaseBackend):
+    def __init__(self, engine_bytes, width, height, device_id):
+        super().__init__()
+        gpu_devide_id = int(device_id[4:])
+        cuda.init()
+        dev = cuda.Device(gpu_devide_id)
+        self.ctx = dev.make_context()
```

Comment:
Can you rename `self.ctx` to `self.context`? It's not a very long name as it is

Reply:
ctx is the cuda context
context is tensorrt context
I'll rename those variables to make them clear

Reply:
Sounds good. Spelling them out would probably be fine, `cuda_context` and `tensorrt_context` still isn't too long.